[Tooltip] Add placement `followCursor` (#22876)
mui-org/material-ui: docs/src/pages/components/tooltips/tooltips.md

{{"demo": "pages/components/tooltips/TransitionsTooltips.js"}}

## Follow cursor

> You can enable the tooltip to follow the cursor with a single prop:

Reviewer: ```suggestion
You can enable the tooltip to follow the cursor by setting `followCursor={true}`.
```

{{"demo": "pages/components/tooltips/FollowCursorTooltips.js"}}

## Faked reference object

In the event you need to implement a custom placement, you can use the `anchorEl` prop:
The value of the `anchorEl` prop can be a reference to a fake DOM element.

> You just need to create an object shaped like the [`ReferenceObject`](https://github.com/FezVrasta/popper.js/blob/0642ce0ddeffe3c7c033a412d4d60ce7ec8193c3/packages/popper/index.d.ts#L118-L123).

Reviewer: ```suggestion
You need to create an object shaped like the [`ReferenceObject`](https://github.com/FezVrasta/popper.js/blob/0642ce0ddeffe3c7c033a412d4d60ce7ec8193c3/packages/popper/index.d.ts#L118-L123).
```
